apsuite/commissioning_scripts/measure_idffwd.py

```diff
@@ -200,15 +196,31 @@ def _init_ffwd_table(self):
 
     def _static_get_trajectory(self):
         """."""
+        sofb = self.devices['sofb']
+
         # reset SOFB buffer and wait for filling
-        self.devices['sofb'].cmd_reset()
-        _time.sleep(2 * self.params.sofb_nrpts * 1/MeasAPUFFWD._SOFB_FREQ)
+        sofb.cmd_reset()
+        wait_factor = 1 + self.params.sofb_overwait/100
+        wait_nominal = self.params.sofb_nrpts * 1/MeasAPUFFWD._SOFB_FREQ
```

Comment:
Instead of doing this algebra you could reset sofb buffer and use an already implemented method of the SOFB device to wait there buffer to be filled.